app/helpers.php

```diff
@@ -251,6 +251,11 @@ function cast(mixed $source, string $classOrType, array $constructParams = [], a
             return Cast::castEnum($source, $classOrType);
         }
 
+        // Convert string to date if required.
+        if (is_string($source) && Carbon::class === $classOrType) {
+            return new Carbon($source);
```

Comment:
I know my PR (https://github.com/Leantime/leantime/pull/2386) is still outstanding but we should cast this to DateTimeHelper (which inherits from CarbonImmutable). That way all of the Carbon Macros are available

Reply:
Cast to DateTimeHelper in https://github.com/Leantime/leantime/pull/2398/commits/53f3ec745c0e01c4d232a28795e99f21f15c2c96.

Reply:
Wait, that does not work.

Reply:
I've rebased my branch to branch out from https://github.com/Leantime/leantime/tree/timesheet-Refactor to be able to build on your changes.